gobblin-utility/src/main/java/gobblin/util/JobLauncherUtils.java

```diff
@@ -71,27 +71,30 @@ public static String newMultiTaskId(String jobId, int sequence) {
   /**
    * Cleanup staging data of a Gobblin task.
    *
-   * @param taskState task state
+   * @param state workunit state
    */
-  public static void cleanStagingData(State taskState, Logger logger) throws IOException {
-    int branches = taskState.getPropAsInt(ConfigurationKeys.FORK_BRANCHES_KEY, 1);
+  public static void cleanStagingData(WorkUnitState state, Logger logger) throws IOException {
+    int branches = state.getPropAsInt(ConfigurationKeys.FORK_BRANCHES_KEY, 1);
     for (int i = 0; i < branches; i++) {
-      String writerFsUri = taskState
+      String writerFsUri = state
           .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, branches, i),
               ConfigurationKeys.LOCAL_FS_URI);
       FileSystem fs = FileSystem.get(URI.create(writerFsUri), new Configuration());
 
-      String writerFilePath = taskState
+      String writerFilePath = state
           .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_PATH, branches, i));
+
+      // If the WRITER_FILE_PATH has not been set, use the default value
       if (Strings.isNullOrEmpty(writerFilePath)) {
-        // The job may be cancelled before the task starts, so this may not be set.
-        continue;
+        writerFilePath =
+            state.getExtract().getOutputFilePath() + Path.SEPARATOR
```

Comment:
We need a helper class to construct the staging directory path which should be used by this class and the Fork operator .